content/hoon/lagoon/build.md

```diff
@@ -0,0 +1,362 @@
+## `+eye` {#eye}
+
+Identity matrix of shape nxn.
+
+#### Accepts
+
+A `$meta`
+
+#### Produces
+
+A `$ray`
+
+#### Source
+
+```hoon
+++  eye      ::  produces identity matrix of shape nxn.
+  |=  =meta
+  ^-  ray
+  ~_  leaf+"lagoon-fail"
+  ?>  =(2 (lent shape.meta))
+  ?>  =((snag 0 shape.meta) (snag 1 shape.meta))
+  =/  n  (snag 0 shape.meta)
+  =<  +
+  %^    spin
+      (gulf 0 (dec n))
+    ^-  ray  (zeros [~[n n] bloq.meta kind.meta ~])
+  |=  [i=@ r=ray]
+  :: [i (set-item r ~[i i] 1)]
+  :-  i
+  %^  set-item
+      r
+    ~[i i]
+  ^-  @
+  ?-    kind.meta
+      %uint  `@`1
+    ::
+      %int2  !!
+    ::
+      %i754
+    ?+  bloq.meta  ~|(bloq.meta !!)
+      %7  .~~~1
+      %6  .~1
+      %5  .1
+      %4  .~~1
+    ==
+  ==
+```
+
+---
+
+## `+zeros` {#zeros}
+
+Zeroes
+
+#### Accepts
+
+A `$meta` of the target shape and parameters
+
+#### Produces
+
+A `$ray`
+
+#### Source
+
+```hoon
+++  zeros
+  |=  =meta  ^-  ray
+  ~_  leaf+"lagoon-fail"
+  :-  meta
+  (lsh [bloq.meta (roll shape.meta ^mul)] 1)
+```
+
+---
+
+## `+ones` {#ones}
+
+Ones
+
+#### Accepts
+
+A `$meta` of the target shape and parameters
+
+#### Produces
+
+A `$ray`
+
+#### Source
+
+```hoon
+++  ones
+  |=  =meta  ^-  ray
+  ~_  leaf+"lagoon-fail"
+  =/  one
+    ?-    kind.meta
+        %uint  `@`1
+      ::
+        %int2  !!
+      ::
+        %i754
+      ?+  bloq.meta  !!
+        %7  .~~~1
+        %6  .~1
+        %5  .1
+        %4  .~~1
+      ==
+    ==
+  (fill meta one)
+```
+
+---
+
+## `+iota` {#iota}
+
+Produce a 1-dimensional index array. Only produces %uint. Note that this runs from 0 to n-1.  (The point of ++iota is to be an index, so it needs to pattern-match the context rather than slavishly follow APL.)
```

Comment:
Should be formatted like so

> Produce a 1-dimensional index array. Only produces `%uint`. Note that this runs from 0 to $$n-1$$. (The point of `+iota` is to be an index, so it needs to pattern-match the context rather than slavishly follow APL.)